### PR TITLE
docs: FIFO SNS and SQS Release 1.22

### PR DESCRIPTION
**Pull request by paragbhingre**
Release 1.22 docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: site/content/blogs/release-v122.en.md [hunk 1]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
 ## IAM Role Permissions Boundary
 
 ## FIFO SNS/SQS
+SNS FIFO Topics and SQS FIFO Queues for your FIFO publish subscribe architecture is now available using Copilot.
+
+####You can configure your manifest to have SNS FIFO Topics enabled for your services.
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md]: Maybe use `###`?
```suggestion
###You can configure your manifest to have SNS FIFO Topics enabled for your services. 
```

File: site/content/blogs/release-v122.en.md [hunk 2]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
 ## IAM Role Permissions Boundary
 
 ## FIFO SNS/SQS
+SNS FIFO Topics and SQS FIFO Queues for your FIFO publish subscribe architecture is now available using Copilot.
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md]:
```suggestion
SNS FIFO Topics and SQS FIFO Queues for your FIFO publish subscribe architecture is now available in Copilot.
```

File: site/content/blogs/release-v122.en.md [hunk 3]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+
+####You can configure your manifest to have SNS FIFO Topics enabled for your services. 
+
+You can specify `fifo: true` in your environment manifest under `publish.topics` as shown below and Copilot will create FIFO SNS Topic for you with the default settings given by the CFN.
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md]:
```suggestion
You can specify `fifo: true` in your environment manifest under `publish.topics` as shown below and Copilot will create a FIFO SNS Topic for you.
```

File: site/content/blogs/release-v122.en.md [hunk 4]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+
+```yaml
+publish:
+  topics:
+    - name: mytopic
+      fifo: true
+```
+
+Alternatively, you can also specify the advanced FIFO SNS Topic configuration as:
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md]: Shall we briefly explain what `content_based_deduplication` does?

File: site/content/blogs/release-v122.en.md [hunk 5]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+```yaml
+publish:
+  topics:
+    - name: mytopic
+      fifo:
+        content_based_deduplication: true
+```
+
+#### SQS FIFO Queues are also configurable for your worker service.
+You can specify `fifo: true` in your environment manifest under `subscribe.topics.queue` or `subscribe.queue` as shown below and Copilot will create FIFO SQS Queue for you with the default settings given by the CFN.
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md]:
```suggestion
You can specify `fifo: true` in your environment manifest under `subscribe.topics.queue` or `subscribe.queue` as shown below and Copilot will create a FIFO SQS Queue for you.
```

File: site/content/blogs/release-v122.en.md [hunk 6]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+subscribe:
+  topics:
+    - name: mytopic
+      service: myservice
+      queue: 
+        fifo: true // topics specific SQS FIFO Queue
+  queue:
+    fifo: true // default SQS FIFO Queue
+```
+Alternatively, you can also specify the advanced FIFO SQS Queue configuration as:
```
Comment (iamhopaul123) [inline site/content/blogs/release-v122.en.md:89]: Same feedback as above. As a blog post I think it is better to go through scenarios: for default scenario vs another specific scenario that default setting can't satisfy them (don't necessarily include every possible advanced fields).

File: site/content/blogs/release-v122.en.md (hunk 4 above)
Comment (paragbhingre) [reply site/content/blogs/release-v122.en.md]: I thought of explaining that but then we have 3 more new fields and we need to explain everything in that case. And we are anyway explaining all the fields in the Copilot docs, so they can take a look at there for more information.

File: site/content/blogs/release-v122.en.md (hunk 6 above)
Comment (paragbhingre) [reply site/content/blogs/release-v122.en.md:89]: Actually that is what I am doing here, if you look at the example then you will see I am showing usage of all the new fields in the example.

File: site/content/blogs/release-v122.en.md (hunk 1 above)
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]: nit:
```suggestion
### You can configure your manifest to have SNS FIFO Topics for your services. 
```

File: site/content/blogs/release-v122.en.md (hunk 3 above)
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]: nit:
```suggestion
You can specify `fifo: true` in the service manifest for your topics. Copilot will create FIFO SNS Topics for you with the default settings given by the CFN.
```

File: site/content/blogs/release-v122.en.md (hunk 2 above)
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]: I think it'd probably be a good idea for us to explain why they'd want to use fifo:

```suggestion
To have strict message ordering and message deduplication for your publish/subscribe architecture, you can now use SNS FIFO Topics and SQS FIFO Queues.
```

File: site/content/blogs/release-v122.en.md [hunk 7]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
 ## IAM Role Permissions Boundary
 
 ## FIFO SNS/SQS
+SNS FIFO Topics and SQS FIFO Queues for your FIFO publish subscribe architecture is now available in Copilot.
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
SNS FIFO Topics and SQS FIFO Queues for your publish-subscribe architecture is now available in Copilot.
```

File: site/content/blogs/release-v122.en.md [hunk 8]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+
+###You can configure your service manifest to have SNS FIFO Topics enabled for your services.
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
### SNS FIFO topics for services
```

File: site/content/blogs/release-v122.en.md [hunk 9]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
 ## IAM Role Permissions Boundary
 
 ## FIFO SNS/SQS
+SNS FIFO Topics and SQS FIFO Queues for your FIFO publish subscribe architecture is now available in Copilot.
+
+###You can configure your service manifest to have SNS FIFO Topics enabled for your services. 
+
+You can specify `fifo: true` in your service manifest under `publish.topics` as shown below and Copilot will create a FIFO SNS Topic for you.
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
You can specify `fifo: true` in your service manifest under `publish.topics` as shown below and Copilot will create a FIFO SNS topic for you.
```

File: site/content/blogs/release-v122.en.md [hunk 10]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+Alternatively, you can also specify the advanced FIFO SNS Topic configuration as:
+```yaml
+publish:
+  topics:
+    - name: mytopic
+      fifo:
+        content_based_deduplication: true
+```
+
+### SQS FIFO Queues are also configurable for your worker service.
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
### SQS FIFO queues for worker services
```

File: site/content/blogs/release-v122.en.md [hunk 11]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+You can specify `fifo: true` in your worker service manifest under `subscribe.topics.queue` or `subscribe.queue` as shown below and Copilot will create a FIFO SQS Queue and Subscriptions for you.
+
+```yaml
+subscribe:
+  topics:
+    - name: mytopic
+      service: myservice
+      queue: 
+        fifo: true // topics specific SQS FIFO Queue
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
        fifo: true # topics specific SQS FIFO Queue
```

File: site/content/blogs/release-v122.en.md [hunk 12]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+  queue:
+    fifo: true // default SQS FIFO Queue
```
Comment (dannyrandall) [inline site/content/blogs/release-v122.en.md]:
```suggestion
    fifo: true # default SQS FIFO Queue
```

File: site/content/blogs/release-v122.en.md [hunk 13]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+#### SQS FIFO Queues are also configurable for your worker service.
+You can specify `fifo: true` in your environment manifest under `subscribe.topics.queue` or `subscribe.queue` as shown below and Copilot will create FIFO SQS Queue for you with the default settings given by the CFN.
+
+```yaml
+subscribe:
+  topics:
+    - name: mytopic
+      service: myservice
+      queue: 
+        fifo: true // topics specific SQS FIFO Queue
```
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]:
```suggestion
        fifo: true # Configure the dedicated queue for "mytopic" to be FIFO.
```

File: site/content/blogs/release-v122.en.md [hunk 14]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+  queue:
+    fifo: true // default SQS FIFO Queue
```
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]:
```suggestion
    fifo: true # Configure the default SQS queue to be FIFO.
```

File: site/content/blogs/release-v122.en.md (hunk 3 above)
Comment (paragbhingre) [reply site/content/blogs/release-v122.en.md]: I have alternative change [mentioned](https://github.com/aws/copilot-cli/pull/4025/commits/0bb0364ebdc7c087c3089664c973db05665e9a52#diff-04ecc586aa589f72329c52799921229fc5ff7e898637bf1c965f2ea4a15a7199R42) here. Do let me know if that works for you?

File: site/content/blogs/release-v122.en.md [hunk 15]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
 ## IAM Role Permissions Boundary
 
 ## FIFO SNS/SQS
+To have strict message ordering and message deduplication for your publish-subscribe architecture, you can now use SNS FIFO Topics and SQS FIFO Queues.
+
+### You can configure your manifest to have SNS FIFO Topics for your services.
+
+You can specify `fifo: true` in your service manifest under `publish.topics` as shown below and Copilot will create a FIFO SNS topic for you.
```
Comment (efekarakus) [inline site/content/blogs/release-v122.en.md]: nit: (same below)
```suggestion
You can specify `fifo: true` in your service manifest under `publish.topics` as shown below and Copilot will create a SNS FIFO topic for you.
```

File: site/content/blogs/release-v122.en.md [hunk 16]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+
+Alternatively, you can also specify the advanced FIFO SNS Topic configuration as:
+```yaml
+publish:
+  topics:
+    - name: mytopic
+      fifo:
+        content_based_deduplication: true
+```
+
```
Comment (efekarakus) [inline site/content/blogs/release-v122.en.md:73]: Can we add links to the manifest specification at the bottom of the SNS and SQS sections, for example:
```
 For the full specification of FIFO topics, see [manifest specification](../docs/manifest/lb-web-service.en.md#theFifoID😅)
 
  For the full specification of FIFO queues, see [manifest specification](../docs/manifest/worker-service.en.md#theFifoID😅)
 ```

**Review by efekarakus — COMMENTED** [review]
Looks great! just a tiny request

File: site/content/blogs/release-v122.en.md (hunk 3 above)
Comment (Lou1415926) [reply site/content/blogs/release-v122.en.md]: :shipit

File: site/content/blogs/release-v122.en.md [hunk 17]
```diff
@@ -35,6 +35,58 @@ Copilot v1.22 brings several new features and improvements:
...
+```yaml
+subscribe:
+  topics:
+    - name: mytopic
+      service: myservice
+      queue: 
+        fifo: true # topics specific SQS FIFO Queue
+  queue:
+    fifo: true # default SQS FIFO Queue
```
Comment (Lou1415926) [inline site/content/blogs/release-v122.en.md]: I think we *might* want to be clear that this is configuring the one and only one default queue to be FIFO, not a default queue for the fifo topics (meaning that there could be a default standard queue along side which is not the case). 

Besides we can probably lower-case "Queue"..?

**Review by efekarakus — APPROVED** [review]

**Review by Lou1415926 — APPROVED** [review]

**Pull request merged**